include/natalie/process_module.hpp

```diff
@@ -96,6 +144,66 @@ class ProcessModule : public Object {
         }
         return gid;
     }
+
+    static int value_to_resource(Env *env, Value val) {
+        int resource;
+        auto to_str = "to_str"_s;
+        auto to_int = "to_int"_s;
+        SymbolObject *rlimit_symbol = nullptr;
+        if (val->is_symbol() || val->is_string() || val->respond_to(env, to_str)) {
+            if (val->is_symbol()) {
+                rlimit_symbol = val->as_symbol();
+            } else if (val->is_string()) {
+                rlimit_symbol = val->as_string()->to_symbol(env);
+            } else {
+                auto tsval = val->send(env, to_str);
+                if (tsval->is_string()) {
+                    rlimit_symbol = tsval->as_string()->to_sym(env)->as_symbol();
+                }
+            }
+        }
+        if (rlimit_symbol) {
+            rlimit_symbol->assert_type(env, Object::Type::Symbol, "Symbol");
+            StringObject *rlimit_str = new StringObject { "RLIMIT_" };
+            rlimit_str->append(rlimit_symbol->string());
+            rlimit_symbol = rlimit_str->as_string()->to_symbol(env);
+            auto ProcessMod = GlobalEnv::the()->Object()->const_fetch("Process"_s)->as_module();
+            Value rlimval = ProcessMod->const_get(rlimit_symbol);
+            if (!rlimval || !rlimval->is_integer()) {
+                env->raise("ArgumentError", "invalid resource {}", rlimit_symbol->string());
+            }
+            val = rlimval->as_integer();
+        }
+
+        if (!val->is_integer() && val->respond_to(env, to_int)) {
+            val = val->send(env, to_int);
+        }
+
+        val->assert_type(env, Object::Type::Integer, "Integer");
+        resource = IntegerObject::convert_to_nat_int_t(env, val);
+        return resource;
+    }
+
+    /*
+        if (resstr == "AS") { return RLIMIT_AS; }
+        if (resstr == "CORE") { return RLIMIT_CORE; }
+        if (resstr == "CPU") { return RLIMIT_CPU; }
+        if (resstr == "DATA") { return RLIMIT_DATA; }
+        if (resstr == "FSIZE") { return RLIMIT_FSIZE; }
+        if (resstr == "MEMLOCK") { return RLIMIT_MEMLOCK; }
+        if (resstr == "MSGQUEUE") { return RLIMIT_MSGQUEUE; }
+        if (resstr == "NPROC") { return RLIMIT_NPROC; }
+        if (resstr == "RLIMIT_NICE") { return RLIMIT_NICE ; }
+        if (resstr == "RLIMIT_NOFILE") { return RLIMIT_NOFILE ; }
+        //if (resstr == "NPTS") { return RLIMIT_NPTS; }
+        if (resstr == "RSS") { return RLIMIT_RSS; }
+        if (resstr == "RLIMIT_RTPRIO") { return RLIMIT_RTPRIO ; }
+        //if (resstr == "RLIMIT_RTTIME") { return RLIMIT_RTTIME ; }
+        //if (resstr == "RLIMIT_SBSIZE") { return RLIMIT_SBSIZE ; }
+        if (resstr == "RLIMIT_SIGPENDING") { return RLIMIT_SIGPENDING ; }
+        if (resstr == "STACK") { return RLIMIT_STACK; }
+        env->raise("ArgumentError", "invalid resource {}", resstr);
+        */
```

Comment:
What's the story with this commented out code? Is this something we will want for later, or just leftover during debugging? If you want to leave it in, please add a note about what it will be used for.

Reply:
Oops, it should've been removed and was an earlier version before I thought of a better way. I need some sort of notifier when I check in new comments > 2 lines.